packages/bundle-source/src/index.js

```diff
@@ -84,105 +180,36 @@ export default async function bundleSource(
-        loc.end = { ...loc.start };
-        for (const pos of ['start', 'end']) {
-          if (loc[pos]) {
-            const newPos = consumer.originalPositionFor(loc[pos]);
-            if (newPos.source !== null) {
-              lastPos = {
-                line: newPos.line,
-                column: newPos.column,
-              };
-            }
-            loc[pos] = lastPos;
-          }
-        }
-        unmapped.add(loc);
-      };
-    }
-
-    const rewriteComment = node => {
-      node.type = 'CommentBlock';
-      // Within comments...
-      node.value = node.value
-        // ...strip extraneous comment whitespace
-        .replace(/^\s+/gm, ' ')
-        // ...replace HTML comments with a defanged version to pass SES restrictions.
-        .replace(HTML_COMMENT_START_RE, '<!X-')
-        .replace(HTML_COMMENT_END_RE, '-X>')
-        // ...replace import expressions with a defanged version to pass SES restrictions.
-        .replace(IMPORT_RE, 'X$1$2')
-        // ...replace end-of-comment markers
-        .replace(/\*\//g, '*X/');
-      if (unmapLoc) {
-        unmapLoc(node.loc);
+  await Promise.all(
+    output.map(async chunk => {
+      if (chunk.isAsset) {
+        throw Error(`unprepared for assets: ${chunk.fileName}`);
+      }
+      const { code, fileName, isEntry } = chunk;
+      if (isEntry) {
+        entrypoint = fileName;
       }
-      // console.log(JSON.stringify(node, undefined, 2));
-    };
 
-    babelTraverse(ast, {
-      enter(p) {
-        const { loc, leadingComments, trailingComments } = p.node;
-        if (p.node.comments) {
-          p.node.comments = [];
-        }
-        // Rewrite all comments.
-        (leadingComments || []).forEach(rewriteComment);
-        if (p.node.type.startsWith('Comment')) {
-          rewriteComment(p.node);
-        }
-        // If not a comment, and we are unmapping the source maps,
-        // then do it for this location.
-        if (unmapLoc) {
-          unmapLoc(loc);
-        }
-        (trailingComments || []).forEach(rewriteComment);
-      },
-    });
+      const useLocationUnmap =
+        moduleFormat === 'nestedEvaluate' && !fileName.startsWith('_virtual/');
 
-    // Now generate the sources with the new positions.
-    sourceBundle[fileName] = babelGenerate(ast, { retainLines: true }).code;
-    // console.log(`==== sourceBundle[${fileName}]\n${sourceBundle[fileName]}\n====`);
-  }
+      const { code: transformedCode } = await transformSource(code, {
+        sourceMap: chunk.map,
+        useLocationUnmap,
+      });
+      sourceBundles[fileName] = transformedCode;
+
+      // console.log(`==== sourceBundles[${fileName}]\n${sourceBundles[fileName]}\n====`);
+    }),
+  );
 
   if (!entrypoint) {
     throw Error('No entrypoint found in output bundle');
   }
 
-  // 'sourceBundle' is now an object that contains multiple programs, which references
+  // 'sourceBundles' is now an object that contains multiple programs, which references
```

Comment:
I can live with this, but to be pedantic, a set of source programs is called a "bundle".  There is only one bundle in this function, not multiple bundles.